src/docfx/build/moniker/MonikersProvider.cs

```diff
@@ -3,23 +3,44 @@
 
 using System;
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
 
 namespace Microsoft.Docs.Build
 {
     internal class MonikersProvider
     {
         private readonly List<(Func<string, bool> glob, string monikerRange)> _rules = new List<(Func<string, bool>, string)>();
+        private readonly MonikerRangeParser _rangeParser;
+        private readonly Lazy<MonikerComparer> _monikerAscendingComparer;
+        private readonly Lazy<MonikerComparer> _monikerDescendingComparer;
 
-        public MonikersProvider(Config config)
+        public MonikerComparer AscendingComparer => _monikerAscendingComparer.Value;
+
+        public MonikerComparer DescendingComparer => _monikerDescendingComparer.Value;
```

Comment:
a single `Comparer` is enough. There is no `StringDescendingComparer` for example.  The only use of `DescendingComparer` is in `XrefMap` which should be using `OrderByDescending` over with a comparer.